src/resultset/resultset.c

```diff
@@ -68,6 +68,9 @@ static void _ResultSet_ReplayStats(RedisModuleCtx *ctx, ResultSet *set) {
 		RedisModule_ReplyWithStringBuffer(ctx, (const char *)buff, buflen);
 	}
 
+	buflen = sprintf(buff, "Cached execution: %s", set->stats.cached ? "true" : "false");
+	RedisModule_ReplyWithStringBuffer(ctx, (const char *)buff, buflen);
```

Comment:
I don't think the user needs to know this. The cache should be opaque in general, and especially if we retain separate caches per thread the output will just be confusing.